test/k8sdeployment/connection_test.go

```diff
@@ -0,0 +1,43 @@
+package k8sdeployment
+
+import (
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/coredns/ci/test/kubernetes"
+)
+
+// This test restarts the kube-APIserver and checks if CoreDNS connection to the API
+// is valid and there are no failures.
+// This test is to catch bugs/errors such as the one reported in https://github.com/coredns/coredns/issues/2464
+func TestConnectionAfterAPIRestart(t *testing.T) {
+	// Apply manifests via coredns/deployment deployment script ...
+	path := os.Getenv("DEPLOYMENTPATH")
+	cmd := exec.Command("sh", "-c", "./deploy.sh -s -i 10.96.0.10 -r 10.96.0.0/8 -r 172.17.0.0/16 | kubectl apply -f -")
+	cmd.Dir = path + "/kubernetes"
+	cmdout, err := cmd.CombinedOutput()
+	if err != nil {
+		t.Fatalf("deployment script failed: %s\nerr: %s", string(cmdout), err)
+	}
+
```

Comment:
Probably should wait until the coredns pods come up at this point before doing the API test.